### DISCUSSION TITLE
Custom checkbox states are not transformed

### DISCUSSION THREAD

**Discussion by jneidel** [minad/org-modern#243] [Ideas]
**Problem description**

I would like to use the canceled state in my checkboxes.
For that I use `[_]` (ascii code 95).

```lisp
(setq org-modern-checkbox
      '((88 . "") (45 . "") (32 . "×") (95 . "") (44 . "u") (46 . ":")))
```

44 (,) and 46 (.) were added as other arbitrary examples.

rendered icons:
![screenshot_24-11-14_19:22:23](https://github.com/user-attachments/assets/b2c18bf3-7f6e-4c6d-bfe4-b274398ebfdf)

org rendering:
![screenshot_24-11-14_19:25:57](https://github.com/user-attachments/assets/d9500dad-bc90-48fc-ad3a-412c352d1dd2)

customize:
<img width=400px src="https://github.com/user-attachments/assets/34ac63d0-c8e0-4e7f-9114-b851a93b8eb9"/>

**Expected behavior**

Have the checkbox with the char `_`, `,` or `.` transformed.

**To reproduce**

in the config
```lisp
(setq org-modern-checkbox
      '((44 . "u") (46 . ":")))
```

in an org file
```org
- [,] test1
- [.] test2
```

**Comment by minad** [comment]
Org-modern mainly targets the standard Org functionality. I wonder if Org supports custom check box states in any way? In my tests only `[ ]` and `[X]` are highlighted by Org.

**Reply by jneidel** [reply]
In progress also [-]: `C-u C-u C-c C-x C-b` marks a checkbox as such.

**Reply by jneidel** [reply]
Even if org ignores my custom states, they would still be useful to me. The meaning of canceled is not otherwise covered.

**Reply by minad** [reply]
Sure I understand. But I have to draw a line somewhere regarding the feature set. This is out of scope of org-modern then. My suggestion would be to propose custom checkbox states to Org itself - I would also find them useful. After their addition in Org, I will also support them here.
